packages/smooth_app/lib/pages/product/common/product_query_page.dart

```diff
@@ -410,34 +408,24 @@ class _ProductQueryPageState extends State<ProductQueryPage>
         ),
       );
 
-  _Action _getRefreshAction(
-    final AppLocalizations appLocalizations,
-  ) =>
-      _Action(
-        text: appLocalizations.label_refresh,
-        iconData: Icons.refresh,
-        onPressed: () async {
-          final bool? success =
-              await _loadAndRefreshDisplay(_model.loadFromTop());
-          if (success == true) {
-            _scrollToTop(instant: true);
-          }
-        },
-      );
-
   void retryConnection() =>
       setState(() => _model = _getModel(widget.productListSupplier));
 
   ProductQueryModel _getModel(final ProductListSupplier supplier) =>
       ProductQueryModel(supplier);
 
   Future<void> refreshList() async {
-    final ProductListSupplier? refreshSupplier =
-        widget.productListSupplier.getRefreshSupplier();
-    setState(
-      // How do we refresh a supplier that has no refresher? With itself.
-      () => _model = _getModel(refreshSupplier ?? widget.productListSupplier),
-    );
+    bool successfullyLoaded = false;
+    try {
+      successfullyLoaded = await _model.loadFromTop();
+    } catch (e) {
+      //Add snackbar to show error?
```

Comment:
Let's say that you've fixed the bug (= no refresh).

Currently there's no way for the user to know if it failed or not, so please put back:
```dart
await LoadingDialog.error(
         context: context,
         title: _model.loadingError,
       );
```

And create an issue about "uniformisation of pull-to-refresh error", to check (later) how consistent we are in the app.

Reply:
Yes, I agree with you, I'll add this dialog box.